src/Notifier.js

```diff
@@ -0,0 +1,327 @@
+          icon: wfcdLogoUrl,
+        },
+      };
+    case 'fissure':
+      return {
+        head: 'New Fissure Detected',
+        body: {
+          body: `${data.tier} ${data.missionType} • ${data.node}`,
+          icon: wfcdLogoUrl,
+        },
+        type: 'fissure',
+      };
+    case 'news':
+      return {
+        head: 'Warframe Hub',
+        body: {
+          body: data.message,
+          icon: wfcdLogoUrl,
+        },
+        link: data.link,
+        type: 'news',
+      };
+    case 'invasions':
+      var rewards = `${data.attackerReward.asString.length ? `${data.attackerReward.asString} vs ` : ''}${data.defenderReward.asString}`;
+
+      return {
+        head: rewards,
+        body: {
+          body: `${data.desc} • ${data.node}\n${data.eta.replace('-Infinityd', '??').replace('Infinityd', '??')} Remaining`,
+          icon: wfcdLogoUrl,
+        },
+        sound: 'drum',
+        type: 'invasion',
+      };
+    case 'vallis.warm':
+      return {
+        head: 'Vallis - Get your Warm Fishing on!',
+        body: {
+          body: data.shortString,
+          icon: wfcdLogoUrl,
+        },
+        type: 'warm',
+      };
+    case 'vallis.cold':
+      return {
+        head: 'Vallis - Back to the Cold!',
+        body: {
+          body: data.shortString,
+          icon: wfcdLogoUrl,
+        },
+        type: 'cold',
+      };
+    case 'nightwave':
+      return {
+        head: `Nightwave - ${data.title}`,
+        body: {
+          body: `${data.daily ? 'Daily: ' : (data.isElite ? 'Weekly Elite: ' : 'Weekly: ')}${data.desc}`,
+          icon: wfcdLogoUrl,
+        },
+      };
+    default:
+      return defaultNotificationBody;
+  }
+};
+
+class Notifier {
+  constructor (store) {
+    this.store = store;
+    this.notifier = Vue.notification;
+  }
+
+  isNotifiable(id, event, items) {
+    const currentNotifieds = this.store.getters.notifiedIds;
+    const eventGood = !currentNotifieds.includes(id) && this.trackedEvents.includes(event);
+    const includesItems = items && items.length ? this.trackedRewards.some((r) => items.indexOf(r) >= 0) : true;
+    return eventGood && includesItems;
+  }
+
+  async notify (notifications) {
+    const usePush = true || !document.hasFocus();
+    notifications.forEach((notification) => {
+      if (usePush) {
+        this.notifier.show(notification.head, notification.body, {
+          onclick: (event) => {
+            if (notification.link) {
+              event.preventDefault();
+              window.open(notification.link, '_blank');
+            }
+          }
+        });
+
+        if (this.store.getters.sounds.includes(notification.type)) {
+          if (notification.sound === 'drum') {
+            const audio = new Audio(drum);
+            audio.volume = 0.2;
+            audio.play();
+          }
+          if (notification.sound === 'eidolon') {
```

Comment:
This is an opportunity for a future refactor involving the factory pattern.

Reply:
hmmm, if there was more than 2 sounds, that would be more useful, but there's only 2 at this point.